### ISSUE TITLE
Always crash during COPY FROM in cluster mode

### ISSUE THREAD

**Issue by kszumilak** [crate/crate#6339]
**CrateDB version**:
2.1.7

**JVM version**:
java version "1.8.0_144"
Java(TM) SE Runtime Environment (build 1.8.0_144-b01)
Java HotSpot(TM) 64-Bit Server VM (build 25.144-b01, mixed mode)

**OS version / environment description**:
3-8 nodes on Azure
56GB RAM, 16 cores, 2TB ssd each
Linux 4.11.0-1011-azure #11-Ubuntu SMP Tue Sep 19 19:03:54 UTC 2017 x86_64 GNU/Linux

**Problem description**:
During COPY FROM, a node always dies (different nodes). The job is killed. Various errors, for example:
[hs_err_pid2734.log](https://github.com/crate/crate/files/1344118/hs_err_pid2734.log)
[hs_err_pid3268.log](https://github.com/crate/crate/files/1344119/hs_err_pid3268.log)

**Steps to reproduce**:

Data is on one node.

```
CREATE TABLE actions_16 (
   event_uuid STRING,
   business_profile_id INT,
   create_date TIMESTAMP,
   source STRING,
   action STRING,
   client_id STRING,
   params OBJECT (dynamic)
)
PARTITIONED BY (action)
WITH (number_of_replicas = 0);

ALTER TABLE actions_16 SET (refresh_interval = 0);

SET GLOBAL TRANSIENT indices.store.throttle.type = 'none';

COPY actions_16  PARTITION (action = 'page.visit') FROM '/data/data01/events_gzip/124/data/partitioned/page.visit__*.gz' WITH (compression = 'gzip', bulk_size = 10000);
```

sample input record:
```
{
   "create_date":"1501234567890",
   "source":"client.web.browser.page",
   "params":{
      "browserver":"12.1.1234.123456789",
      "campaign":"0",
      "encoding":"UTF-8",
      "color":"1234-bit",
      "ip":"xxxxxxxxxxxxxxxxxxxxxxxxxxx",
      "userAgent":"Mozilla/5.0 (Windows NT 6.2; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36",
      "s_visits":"6",
      "eventUUID":"12345678-1234-1234-1234-123456789012",
      "ver":"3",
      "uuid":"12345678-1234-1234-1234-123456789012",
      "tablet":false,
      "title":"XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
      "res":"1234x123",
      "long":"12",
      "current":"1501234567",
      "source":"WEB_DESKTOP",
      "init":"1501234567",
      "tracker":"XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
      "user_hash":"",
      "ssuid":"XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
      "ref":"/xxxxxxxxx/xxxxxxxxxxxxxxx",
      "location":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
      "refdomain":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
      "visits":"12",
      "u24_visits":"1",
      "host":"xxxxxxxxxxxxxxxxxxxxxxxx",
      "ogTags":{
         "og:url":"xxxxxxxxxxxxxxxxxxxxxxxxx",
         "og:image":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
         "og:site_name":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
         "og:title":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
      },
      "lang":"xx",
      "last":"1501234567.89",
      "mobile":"false",
      "referrer":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
      "ecookies":"true",
      "uri":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
      "osver":"8",
      "time":1501234567890,
      "productRef":"null",
      "os":"Windows",
      "browser":"Chrome"
   },
   "client_id":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
   "business_profile_id":"1234",
   "event_uuid":"12345678-1234-1234-1234-123456789012"
}
```

Diagnostic queries:
![crate_diag](https://user-images.githubusercontent.com/32389299/31014158-ce1f97e6-a519-11e7-96dc-8042b1394354.png)
![crate_diag2](https://user-images.githubusercontent.com/32389299/31014159-ce20b37e-a519-11e7-9c6b-5987b690c56c.png)

**Comment by joemoe** [comment]
hi @kszumilak, thanks for reporting that. we will look into that.

**Comment by kszumilak** [comment]
We also tried to change GC to G1, but also observed a crash:
[hs_err_pid18101.log](https://github.com/crate/crate/files/1344390/hs_err_pid18101.log)

**Comment by matriv** [comment]
@kszumilak Please don't use G1 gc. There is an open Lucene bug that can lead even to data corruption. From https://wiki.apache.org/lucene-java/JavaBugs:

Do not, under any circumstances, run Lucene with the G1 garbage collector. Lucene's test suite fails with the G1 garbage collector on a regular basis, including bugs that cause index corruption. There is no person on this planet that seems to understand such bugs (see https://bugs.openjdk.java.net/browse/JDK-8038348, open for over a year), so don't count on the situation changing soon. This information is not out of date, and don't think that the next oracle java release will fix the situation.

**Comment by matriv** [comment]
@kszumilak Sorry, please ignore the previous comment, as the issue has been resolved in Java b137.

**Comment by kszumilak** [comment]
@matriv no worries, we were just trying G1 to see if that helps with our problem but no luck with that.

**Comment by kszumilak** [comment]
@matriv @joemoe 
We have narrowed down the issue:
- we are observing the JVM crashes with "4.11.0-1011-azure #11-Ubuntu"  kernel
- we are NOT observing the crashes with the same dataset and same setup on bare metal servers, running a generic fresh kernel
- we are NOT observing the crashes on the same Azure cluster with a generic linux kernel (without Azure modifications)

It seems that whatever modifications the Azure kernel contains are causing JDK to explode during `COPY`.

**Comment by joemoe** [comment]
ok @kszumilak, this is indeed an interesting insight.

**Comment by matriv** [comment]
@kszumilak I have been trying to reproduce the issue on my dev machine, on our dev cluster and also on Azure (Ubuntu - same kernel) with no success so far. Some questions for you:

- The data file that you use, does it reside only in one of the nodes your cluster or is it split and exists in more than one?

- The node that "dies" is the one that has the file, or can be any node of the cluster?

- Is it possible to send the real data file to `marios [at] crate.io` to help us reproduce the issue?

**Comment by mfussenegger** [comment]
> we are NOT observing the crashes with the same dataset and same setup on bare
> metal servers, running a generic fresh kernel

> We are NOT observing the crashes on the same Azure cluster with a generic linux
> kernel (without Azure modifications)

Is the JDK version in all cases the same?

**Comment by kszumilak** [comment]
@matriv 
- the data resides only in one of the nodes of the cluster
- the 'dying' node can by any node of the cluster
- unfortunately we cannot send the data as it holds sensitive information

@mfussenegger 
- yes, the JDK version in all cases is the same

**Comment by jodok** [comment]
for the record: i've also opened a support ticket on our azure account and try to get in contact via our azure experts.

**Comment by jodok** [comment]
we have an update from azure support:

> Thanks for your patience. We received an update from our internal engineering team mentioning that the bug has been escalated and will be resolved in next release of kernel version 4.11.0-1013, which is currently not available. It takes some time for testing, which is expected to release by October 30th .

**Comment by jodok** [comment]
@kszumilak it seems kernel 4.11.0-1013-azure is available: https://packages.ubuntu.com/xenial-updates/linux-image-4.11.0-1013-azure
please let us know if the error still persists!
